packages/shared/lib/sdk/sync.unit.test.ts

```diff
@@ -6,12 +6,45 @@ import configService from '../services/config.service.js';
 import type { CursorPagination, LinkPagination, OffsetPagination } from '../models/Proxy.js';
 import { NangoAction } from './sync.js';
 import { isValidHttpUrl } from '../utils/utils.js';
+import proxyService from '../services/proxy.service.js';
+import type { AxiosResponse } from 'axios';
 
 vi.mock('@nangohq/node', () => {
     const Nango = vi.fn();
     return { Nango };
 });
 
+describe('Proxy', () => {
+    let nangoAction: NangoAction;
+    let nango: Nango;
+    beforeEach(async () => {
+        nangoAction = new NangoAction({
+            secretKey: '***',
+            providerConfigKey: 'github',
+            connectionId: 'connection-1'
+        });
+        nango = new Nango({ secretKey: '***' });
+        (await import('@nangohq/node')).Nango.prototype.getConnection = vi.fn().mockReturnValue({ credentials: {} });
+        vi.spyOn(proxyService, 'route').mockImplementation(() => Promise.resolve({ response: {} as AxiosResponse, activityLogs: [] }));
+    });
+    afterEach(() => {
+        vi.clearAllMocks();
+    });
+
+    it('memoizes connection', async () => {
+        await nangoAction.proxy({ endpoint: '/issues' });
+        await nangoAction.proxy({ endpoint: '/issues' });
+        expect(nango.getConnection).toHaveBeenCalledTimes(1);
+    });
+    it('get connection if memoized connection is too old', async () => {
+        await nangoAction.proxy({ endpoint: '/issues' });
+        const later = Date.now() + 61000;
+        vi.spyOn(Date, 'now').mockReturnValue(later);
```

Comment:
Does the mocking reset after this? if not, it could break other following tests.

Reply:
yep 
```
afterEach(() => {
        vi.clearAllMocks();
    });
```